src/main/java/io/apicurio/datamodels/Library.java

```diff
@@ -136,7 +143,35 @@ public static void visitTree(Node node, IVisitor visitor, TraverserDirection dir
         
         return validator.getValidationProblems();
     }
-    
+
+    /**
+     * Called to validate a data model node.  All validation rules will be evaluated and reported.  The list
+     * of validation problems found during validation is returned.  In addition, validation problems will be
+     * reported on the individual nodes themselves.  Validation problem severity is determined by checking
+     * with the included severity registry.  If the severity registry is null, a default registry is used.
+     * Custom validators can be passed to provide additional validation rules beyond what this Library offers out of the box.
+     *
+     * @param node The document to be validated
+     * @param severityRegistry Supply a custom severity registry. If nothing is passed, the default severity registry will be used
+     * @param extensions Supply an optional list of validation extensions, enabling the use of 3rd-party validators or custom validation rules
+     * @return full list of the validation problems found in the document
+     */
+    public static CompletableFuture<List<ValidationProblem>> validateDocument(Node node, IValidationSeverityRegistry severityRegistry, List<IDocumentValidatorExtension> extensions) {
```

Comment:
@EricWittmann a common way to define a method API in TS/JS would be `validateDocument(Node, ValidationOptions)`, where ValidationOptions would encapsulate any number of options, such as the extension list + severityRegistry.
Is this common in Java? It means we are less likely to introduce backwards incompatible changes in the future if we want to add a new parameter, and you don't have to pass any nulls.